name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/LeewaySearchProvider.java

```diff
@@ -0,0 +1,214 @@
+        {
+            return null;
+        }
+
+        @Override
+        public String getCurrencyCode()
+        {
+            return currencyCode;
+        }
+
+        @Override
+        public String getSource()
+        {
+            return NAME;
+        }
+
+        @Override
+        public boolean hasPrices()
+        {
+            return true;
+        }
+
+        @Override
+        public Security create(ClientSettings settings)
+        {
+            Security security = new Security();
+            security.setName(name);
+            security.setTickerSymbol(symbol);
+            security.setIsin(isin);
+            security.setCurrencyCode(currencyCode);
+            security.setFeed(LeewayQuoteFeed.ID);
+            return security;
+        }
+    }
+
+    private static final String NAME = "PWP Leeway UG"; //$NON-NLS-1$
+    private String apiKey;
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    public void setApiKey(String apiKey)
+    {
+        this.apiKey = apiKey;
+    }
+
+    @Override
+    public List<ResultItem> search(String query, Type type) throws IOException
+    {
+        List<ResultItem> answer = new ArrayList<>();
+
+        // @formatter:off
+        // There are only 50 API/day available, so only query with validated ISIN.
+        // @formatter:on
+
+        if (Isin.isValid(query))
+            addISINSearchPage(answer, query.trim());
+
+        return answer;
+    }
+
+    private void addISINSearchPage(List<ResultItem> answer, String query) throws IOException
+    {
+        String html = new WebAccess("api.leeway.tech", "/api/v1/public/general/isin/" + query) //$NON-NLS-1$ //$NON-NLS-2$
+                        .addParameter("apitoken", apiKey) //$NON-NLS-1$
+                        .get();
+
+        extract(answer, html);
+    }
+
+    /**
+     * Return JSON
+     * {
+     *   "Code": "SAP",
+     *   "Exchange": "XETRA",
+     *   "Name": "SAP SE",
+     *   "Type": "Common Stock",
+     *   "ISIN": "DE0007164600",
+     *   "previousClose": 122.6,
+     *   "previousCloseDate": "2023-05-09",
+     *   "countryName": "Germany",
+     *   "currencyCode": "EUR"
+     * }
+     */
+    
+    @SuppressWarnings("nls")
+    void extract(List<ResultItem> answer, String html)
+    {
+        JSONArray jsonArray = (JSONArray) JSONValue.parse(html);
+
+        if (jsonArray.isEmpty())
+            return;
+
+        for (int i = 0; i < jsonArray.size(); i++)
+        {
+            JSONObject item = (JSONObject) jsonArray.get(i);
+
+            if (item.get("Code") == null || item.get("Exchange") == null || item.get("Name") == null
+                            || item.get("Type") == null || item.get("ISIN") == null || item.get("previousClose") == null
+                            || item.get("previousCloseDate") == null || item.get("countryName") == null
+                            || item.get("currencyCode") == null)
+                continue;
```

Comment:
I propose to move this into the "Result.from" method. That method can already return "null" if no valid item is created. Now you duplicate the "get" logic + you have to places where the code needs to know what properties are available.